src/helpers/agGrid.ts

```diff
@@ -303,12 +350,36 @@ export const executionToAGGridAdapter = (
     const rowHeaders: IGridHeader[]
         = getRowHeaders(dimensions[0].headers, columnDefOptions, makeRowGroups);
 
-    // Add loading indicator to the first column
-    if (addLoadingRenderer && rowHeaders.length > 0) {
-        rowHeaders[0].cellRenderer = addLoadingRenderer;
-    }
-
-    // assign indexes
+    // build sortingMap from resultSpec.sorts
+    const sorting = resultSpec.sorts || [];
+    const sortingMap = {};
+    const { attributeHeaders, measureHeaderItems } = assortDimensionHeaders(dimensions);
+    sorting.forEach((sortItem) => {
```

Comment:
split this new code into at least two functions - for attributes and for measures. It will increase readability

Reply:
done